workers/common/src/main/kotlin/common/env/NetRcManager.kt

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.apoapsis.ortserver.workers.common.context.AuthenticationEvent
+import org.eclipse.apoapsis.ortserver.workers.common.context.AuthenticationListener
+import org.eclipse.apoapsis.ortserver.workers.common.context.WorkerContext
+import org.eclipse.apoapsis.ortserver.workers.common.env.definition.EnvironmentServiceDefinition
+
+import org.slf4j.LoggerFactory
+
+/**
+ * An internal helper class to dynamically update the content of the `.netrc` file when there is change in the
+ * authentication state.
+ *
+ * An instance of this class is created when setting up the environment for executing a worker. It is initialized with
+ * the set of infrastructure services currently available. It is also registered at the central authenticator used by
+ * ORT Server. Its task is to figure out, based on authentication events, which infrastructure services are currently
+ * referenced. These are then written into the `.netrc` file, if they have the corresponding credentials type. This
+ * should prevent conflicts with multiple services for the same host.
+ */
+internal class NetRcManager(
+    /** The current [WorkerContext]. */
+    private val context: WorkerContext,
+
+    /** The currently known set of infrastructure services. */
+    services: Collection<InfrastructureService>
+) : AuthenticationListener {
+    companion object {
+        private val logger = LoggerFactory.getLogger(NetRcManager::class.java)
+
+        /**
+         * Create a new instance of [NetRcManager] and initialize it with the given [context] and [services].
+         */
+        fun create(context: WorkerContext, services: Collection<InfrastructureService>): NetRcManager =
+            NetRcManager(context, services)
+    }
+
+    /** A map with all relevant services using the service name as key. */
+    private val netRcServices = services.filter { CredentialsType.NETRC_FILE in it.credentialsTypes }
+        .associateBy { it.name }
+
+    /**
+     * Stores the services for which authentication events have been received. These are the services that are
+     * currently referenced from the `.netrc` file. The key is the host for which a service is responsible for.
+     */
+    private val authenticatedServices = mutableMapOf<String, EnvironmentServiceDefinition>()
+
+    /** A mutex to guard access to the authenticated services and updates of the `.netrc` file. */
+    private val mutex = Mutex()
+
+    override fun onAuthentication(authenticationEvent: AuthenticationEvent) {
+        logger.info("Received authentication event for service '${authenticationEvent.serviceName}'.")
+
+        netRcServices[authenticationEvent.serviceName]?.also { service ->
+            runBlocking { updateNetRcServices(service) }
+        }
+    }
+
+    /**
+     * Return a new instance of [ConfigFileBuilder] to be used for creating the `.netrc` file.
+     */
+    internal fun createConfigFileBuilder(): ConfigFileBuilder = ConfigFileBuilder(context)
```

Comment:
Unnecessary indirection, just call `ConfigFileBuilder(context)` directly.

Reply:
This is used to inject mocks into test code.